### ISSUE TITLE
Custom render doesn't generate unique key prop

### ISSUE THREAD

**Issue by ark-davegoodchild** [icflorescu/mantine-datatable#399]
**Describe the bug**
Using a custom render for a column, generates the following error:
```
Warning: Encountered two children with the same key, ``. Keys should be unique so that components maintain their identity across updates. Non-unique keys may cause children to be duplicated and/or omitted — the behavior is unsupported and could change in a future version.
```
Even specifying a `key` property on the parent of the returned ReactNode doesn't address the problem.

**To Reproduce**
Create a column with a custom `render` and return any ReactNode

**Expected behavior**
Should not generate the error and should provide a unique key for thw rendered cell, or abide by a key property on the parent of the returned ReactNode

**Screenshots**
N/A

**Desktop (please complete the following information):**

**Smartphone (please complete the following information):**

- Device: macBook Pro 2019 (Intel)
- OS: macOS 12.6
- Browser: Firefox
- Version: 115.0.1

**Additional context**
N/A

**Comment by icflorescu** [comment]
Adding custom render for a column shouldn't generate such an error.
There are many examples in the docs website, such as [this one](https://icflorescu.github.io/mantine-datatable/examples/column-properties), that work properly.

Are you sure that's where the error comes from? Do you have an `id` property on your records, or if you don't, are you [correctly setting the `idAccessor`](https://icflorescu.github.io/mantine-datatable/examples/non-standard-record-ids)?

Can you please share a reproduction?

**Comment by ark-davegoodchild** [comment]
Hi @icflorescu, `idAccessor` is set and used as each record does have a unique ID.

I have verified this on multiple tables. I've removed the `render` and just output the value obtained via the `accessor` and the error goes away. I get an error for every `render` defined (well actually, 2 for each one).

As an aside, I am having to set an empty `accessor` for some of the columns due to it being a type requirement but being rendered with `render`.
